docs/notebooks/demo/navigator/navigator-data-designer-sdk-multi-turn-conversation.ipynb

```diff
@@ -0,0 +1,429 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_designer.special_system_instructions = \"\"\"\n",
+    "    You are an expert conversation designer and domain specialist. Your job is to\n",
+    "    produce realistic user-assistant dialogues for fine-tuning a model. Always ensure:\n",
+    "    - Responses are factually correct and contextually appropriate.\n",
+    "    - Communication is clear, helpful, and matches the complexity level.\n",
+    "    - Avoid disallowed content and toxicity.\n",
+    "    - After the two-turn conversation, provide a single toxicity assessment for the user's messages in the entire conversation.\n",
+    " \"\"\"\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Adding Categorical Seed Columns\n",
+    "\n",
+    "We define categorical seed columns that set the context for the generated dialogues. For example, domain and topic determine what the conversation is about, while complexity guides the level of detail and difficulty. By using `num_new_values_to_generate`, we can automatically expand the range of topics or domains, increasing the diversity of generated data without manually specifying all values."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_designer.add_categorical_seed_column(\n",
+    "    name=\"domain\",\n",
+    "    description=\"The domain of user assistant queries\",\n",
+    "    values=[\"Tech Support\", \"Personal Finances\", \"Educational Guidance\"],\n",
+    "    subcategories=[\n",
+    "        {\n",
+    "            \"name\": \"topic\",\n",
+    "            \"values\": {\n",
+    "                \"Tech Support\": [\n",
+    "                    \"Troubleshooting a Laptop\",\n",
+    "                    \"Setting Up a Home Wi-Fi Network\",\n",
+    "                    \"Installing Software Updates\"\n",
+    "                ],\n",
+    "                \"Personal Finances\": [\n",
+    "                    \"Budgeting Advice\",\n",
+    "                    \"Understanding Taxes\",\n",
+    "                    \"Investment Strategies\"\n",
+    "                ],\n",
+    "                \"Educational Guidance\": [\n",
+    "                    \"Choosing a College Major\",\n",
+    "                    \"Effective Studying Techniques\",\n",
+    "                    \"Learning a New Language\"\n",
+    "                ]\n",
+    "            },\n",
+    "            \"num_new_values_to_generate\": 3\n",
+    "        }\n",
+    "    ],\n",
+    "    num_new_values_to_generate=3\n",
+    ")\n",
+    "\n",
+    "data_designer.add_categorical_seed_column(\n",
+    "    name=\"complexity\",\n",
+    "    description=\"The complexity level of the user query\",\n",
+    "    values=[\"Basic\", \"Intermediate\", \"Advanced\"]\n",
+    ")\n"
```

Comment:
Just a side note: The description only really matters if you are generating new values, since it is injected into the prompt. It's still nice to use it here, though, as a way to make clear what the category represents.

Reply:
Ah good to know that. I thought it may be used in some logging as well.